api/internal/application/auth.go

```diff
@@ -30,7 +31,7 @@ func (l *loginCommandImpl) Handle(ctx context.Context, param LoginParam) (*core.
 	if err != nil {
 		return nil, err
 	}
-	if user.Password != param.Password {
+	if err = password.CheckPasswordHash(user.Password, param.Password); err != nil {
 		return nil, ErrInvalidLogin
 	}
```

Comment:
there is no interface for password hasher?

Reply:
yeah right, I'll add one